### PR TITLE
Add support for http and socket transports

### LINKED ISSUE #64: Transport for eAPI should not be restricted to HTTPS
### Description of Issue/Question

eAPI allows the use of UNIX Sockets, HTTP/HTTPS, and HTTP-Local as various transport options to access the device. It would be nice to have the .open() method take an argument specifying the preferred transport method rather than hardcoding `https` as the only possible mechanism.
### napalm-eos version

(Paste verbatim output from `pip freeze | grep napalm-eos` between quotes below)

```

```
### eOS version

(Paste verbatim output from `show version | json` between quotes below)

```

```
### Error Traceback

(Paste the complete traceback of the exception between quotes below)

```

```

### PR DESCRIPTION
**Pull request by dbarrosop**
Fixes #64 

@narJH27 would you mind testing this branch?

If it works I will document the optional arg in the official documentation.

### CODE REVIEW

File: napalm_eos/eos.py [hunk 1]
```diff
@@ -62,20 +62,33 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
 
         if optional_args is None:
             optional_args = {}
-        self.port = optional_args.get('port', 443)
+
+        self.transport = optional_args.get('eos_transport', 'https')
+
+        if self.transport == 'https':
+            self.port = optional_args.get('port', 443)
+        elif self.transrpot == 'http':
```
Comment (narJH27) [inline napalm_eos/eos.py:70]: Typo: `self.transport` instead of `self.transrpot`

File: napalm_eos/eos.py [hunk 2]
```diff
@@ -62,20 +62,33 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
...
+            self.port = optional_args.get('port', 80)
+
         self.enablepwd = optional_args.get('enable_password', '')
 
     def open(self):
         """Implemantation of NAPALM method open."""
```
Comment (mirceaulinic) [inline napalm_eos/eos.py:76]: I also noticed this one `Implemantation` :)

**Comment by narJH27** [comment]
Also, the `socket` transport can only be used by a script on the box rather than a script trying to connect to the box from the outside. In my testing, my script could connect to the box using a UNIX socket using `pyeapi.connect(<host>)` as opposed to `pyeapi.client.connect()`. Maybe you can look into that?

**Comment by dbarrosop** [comment]
Can you send your script and any relevant configuration? I just want to make sure I test your same exact conditions.

**Comment by narJH27** [comment]
@dbarrosop No relevant conditions as such. But my recollection of using pyeapi with unix sockets was as mentioned above. The script is nothing major, basically opening a connection to the box using the unix socket and calling one of napalm's methods, something like a `get_bgp_neighbors_detail()` which usually takes a few seconds to respond in the normal case.

**Comment by dbarrosop** [comment]
@narJH27 I don't really understand what you are asking for with this:

> the socket transport can only be used by a script on the box rather than a script trying to connect to the box from the outside

Unix sockets are local by nature, you can't connect from the outside unless they are exposed via a TCP port, which is what the http and https transports are for.

**Comment by narJH27** [comment]
@dbarrosop Apologies for using the phrase `from the outside`. What I meant was a local script connecting to the UNIX socket on the box rather than an outside connection using HTTP/HTTPS.

**Comment by dbarrosop** [comment]
So this is done then. Merging.

**Pull request merged**